addons/mass_mailing_sms/static/src/js/fields_sms_widget.js

```diff
@@ -0,0 +1,87 @@
+/** @odoo-module **/
+
+import { _t } from "web.core";
+import SmsWidget from '@sms/js/fields_sms_widget';
+
+const TEXT_URL_REGEX = /https?:\/\/[\w@:%.+&~#=/-]+(?:\?\S+)?/g;  // from tools.mail.TEXT_URL_REGEX
+
+/**
+ * Override to provide extra characters count information to
+ * consider links converted with link_tracker and opt-out
+ * link if the option is selected.
+ */
+SmsWidget.include({
+    events: _.extend({}, SmsWidget.prototype.events || {}, {
+        'focusin': '_onClick',
+    }),
+    /**
+     *@override
+     */
+    willStart: async function () {
+        await this._super.apply(this, arguments);
+        if (this.model === "mailing.mailing" && this.mode === 'edit') {
+            const { optOutLinkCharsCount, linkCharsCount } = await this._rpc({
+                model: 'mailing.mailing',
+                method: 'get_extra_characters_counts',
+                args: [this.res_id]
+            });
+            this.charsCountData = { optOutLinkCharsCount, linkCharsCount };
+            this.noticeLinksReplaced = false;
+        }
+    },
+    /**
+     * Add the characters for opt-out link if enabled.
+     *
+     * @override
+     */
+    _compute: function() {
+        this._super.apply(this, arguments);
+        if (this.charsCountData) {
+            // this is the simplest way to read from another field in a legacy fix.
+            const sms_allow_unsubscribe_el = document.querySelector('div[name="sms_allow_unsubscribe"] input');
+            if (sms_allow_unsubscribe_el && sms_allow_unsubscribe_el.checked) {
+                this.nbrChar += this.charsCountData.optOutLinkCharsCount;
+            }
+        }
+    },
+    /**
+     * Add the applicable extra characters notice.
+     *
+     * @override
+     */
+    _getNbrCharExplanationTemplate: function () {
+        if (this.optOutEnabled) {
+            if (this.noticeLinksReplaced) {
+                return _t("%s characters (including link trackers and opt-out link), fits in %s SMS (%s) ");
+            }
+            return _t("%s characters (including opt-out link), fits in %s SMS (%s) ");
+        }
+        if (this.noticeLinksReplaced) {
+            return _t("%s characters (including link trackers), fits in %s SMS (%s) ");
+        }
+        return this._super.apply(this, arguments);
+    },
+    /**
+     * Convert links to link trackers-length replacements.
+     * Works under the assumption that these trackers urls only
+     * contain GSM7 characters.
+     *
+     * @override
+     */
+    _getValueForSmsCounts: function () {
+        const value = this._super.apply(this, arguments);
+        if (this.charsCountData) {
```

Comment:
thinking of it, can't we return a placeholder for the links ? like 'odoo.com/r/XXX/s/XXX', instead of the char count, so that we can handle "fake links" ?